lib/smb2-cmd-query-directory.c

```diff
@@ -54,7 +54,7 @@ smb2_decode_fileidfulldirectoryinformation(
     struct smb2_iovec *vec)
 {
         uint32_t name_len;
-        uint64_t t;
+        time_t t;
```

Comment:
We can't do this. time_t is not guaranteed to be 64 bit on all platforms and would cause an out of bounds write later in the call to smb2_get_uint64().

Revert this change.

Reply:
👍